src/components/ClusterManager.js

```diff
@@ -547,35 +607,43 @@ export default ajaxCaller(class ClusterManager extends PureComponent {
             [multiple, () => undefined],
             () => 'Update runtime'
           ),
-          onClick: () => this.setState({ open: true }),
+          onClick: () => this.setState({ createModalOpen: true }),
           disabled: isDisabled
         }, [
           div({
             style: { marginLeft: '0.5rem', paddingRight: '0.5rem', color: colors.gray[0] }
           }, [
             div({ style: { fontSize: 12, fontWeight: 'bold' } }, 'Notebook Runtime'),
             div({ style: { fontSize: 10 } }, [
-              span({ style: { textTransform: 'uppercase', fontWeight: 500 } }, currentStatus || 'None'),
+              span({ style: { textTransform: 'uppercase', fontWeight: 500 } }, status || 'None'),
               ` (${Utils.formatUSD(totalCost)} hr)`
             ])
           ]),
           icon('cog', { size: 22, className: 'is-solid', style: { color: isDisabled ? colors.gray[2] : colors.green[0] } })
         ])
       ]),
-      deleting && h(Modal, {
+      deleteModalOpen && h(Modal, {
         title: 'Delete notebook runtime?',
-        onDismiss: () => this.setState({ deleting: false }),
+        onDismiss: () => this.setState({ deleteModalOpen: false }),
         okButton: () => {
-          this.setState({ deleting: false })
+          this.setState({ deleteModalOpen: false })
           this.destroyActiveCluster()
         }
       }, ['Deleting the cluster will stop all running notebooks and associated costs. You can recreate it later, which will take several minutes.']),
-      open && h(NewClusterModal, {
+      createModalOpen && h(NewClusterModal, {
         namespace, currentCluster,
-        onCancel: () => this.setState({ open: false }),
+        onCancel: () => this.setState({ createModalOpen: false }),
         onSuccess: promise => {
-          this.setState({ open: false })
-          this.executeAndRefresh(promise)
+          this.setState({ createModalOpen: false })
+          const doCreate = async () => {
+            if (clusterErrors) {
```

Comment:
There's an edge case here. Since we don't reset `clusterErrors` anywhere, this would cause future create operations to delete the current cluster, even if it wasn't errored. This should probably be checking `status === 'Error'` instead.